src/core/ddsi/src/q_nwif.c

```diff
@@ -39,7 +39,7 @@ static void print_sockerror (const char *msg)
 
 unsigned locator_to_hopefully_unique_uint32 (const nn_locator_t *src)
 {
-  unsigned id;
+  unsigned id = 0;
   if (src->kind == NN_LOCATOR_KIND_UDPv4 || src->kind == NN_LOCATOR_KIND_TCPv4)
     memcpy (&id, src->address + 12, sizeof (id));
   else
```

Comment:
Actually, the else branch shouldn't be IPv6 dependent: it should rather take all bytes & do its thing regardless of the locator type. IPv4 is special in that it fits in a 32-bit int.

Reply:
This one is just to fix a compiler warning about `id` being uninitialized if IPv6 is not available. There's no change in logic. This situation should never arise because non UDPv4 or TCPv4 locators should not be used if nothing else is available on a given target. So the only thing that actually changed is that the result in a scenario that cannot happen is predictable. I propose we leave this as is for now since it is out-of-scope for my work on the FreeRTOS port.